src/client/Wrapper.js

```diff
@@ -73,6 +78,42 @@ export default class Wrapper extends React.PureComponent {
     this.props.getTrendingTopics();
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.locale !== this.props.locale) {
+      this.setState({ locale: this.getAvailableLocale(nextProps.locale) });
+      import(`react-intl/locale-data/${this.getAvailableLocale(nextProps.locale)}`).then((localeData) => {
```

Comment:
I know dynamic imports were not recommended in the past before by the creator of redux - https://github.com/facebookincubator/create-react-app/issues/2176

and he works for facebook, so idk about this one and I'm not familiar with these